### PR TITLE
Added retry to SendAsync when 'Job List Uri' and 400 HTTP code is received

### PR DESCRIPTION
**Pull request by mjanecke**
Added retry to SendAsync when 'Job List Uri' and 400 HTTP code is received

### CODE REVIEW

File: src/Microsoft.DotNet.Helix/JobSender/JobDefinition.cs [hunk 1]
```diff
@@ -242,12 +242,18 @@ public async Task<ISentJob> SendAsync(Action<string> log = null)
...
+                IsRetryableJobListUriHttpError,
                 CancellationToken.None);
 
 
             return new SentJob(JobApi, newJob, resultsStorageContainer?.Uri, string.IsNullOrEmpty(Creator) ? resultsStorageContainer?.ReadSas : string.Empty);
         }
 
+        private static bool IsRetryableJobListUriHttpError(Exception ex)
+        {
+            return ex.Message.Contains("Provided Job List Uri is not accessible") && ex.InnerException is RestApiException raex && (int)raex.Response.StatusCode == 400;
```
Comment (alexperovich) [inline src/Microsoft.DotNet.Helix/JobSender/JobDefinition.cs]: This should also include the check that it uses by default. So we retry on all the other retryable exceptions.

Comment (ChadNedzlek) [inline src/Microsoft.DotNet.Helix/JobSender/JobDefinition.cs]: I wouldn't check the string... we might change this, and the particular strings we return aren't meant to be machine interpreted. Matter of fact, I'm likely going to break this right now.

Is there some sort of error code we should be returning or something?

Comment (mjanecke) [reply src/Microsoft.DotNet.Helix/JobSender/JobDefinition.cs]: TODO: Make an issue to fix this at the Helix API level (and refactor this so we're not retrying based on string content)

Comment (mjanecke) [reply src/Microsoft.DotNet.Helix/JobSender/JobDefinition.cs]: Created https://github.com/dotnet/core-eng/issues/6748 to address this issue.

**Pull request merged**